tket/src/Circuit/macro_circ_info.cpp

```diff
@@ -879,39 +932,54 @@ unit_vector_t Circuit::args_from_frontier(
   EdgeVec ins = get_in_edges(vert);
   unit_vector_t args;
   for (port_t p = 0; p < ins.size(); ++p) {
-    if (get_edgetype(ins[p]) == EdgeType::Boolean) {
-      bool found = false;
-      for (const std::pair<Bit, EdgeVec>& pair :
-           prev_b_frontier->get<TagKey>()) {
-        for (const Edge& edge : pair.second) {
-          if (edge == ins[p]) {
+    switch (get_edgetype(ins[p])) {
+      case EdgeType::WASM: {
+        continue;
+      }
+      case EdgeType::Boolean: {
+        bool found = false;
+        for (const std::pair<Bit, EdgeVec>& pair :
+             prev_b_frontier->get<TagKey>()) {
+          for (const Edge& edge : pair.second) {
+            if (edge == ins[p]) {
+              args.push_back(pair.first);
+              found = true;
+              break;
+            }
+          }
+          if (found) {
+            break;
+          }
+        }
+        if (!found)
+          throw CircuitInvalidity(
+              "Vertex edges not found in CRead frontier. Edge: " +
```

Comment:
```suggestion
              "Vertex edges not found in Boolean frontier. Edge: " +
```

Reply:
done in [suggested changes I](https://github.com/CQCL/tket/pull/743/commits/e06543a3343027c8ff962d77fda395d3b22d9f96)  [e06543a](https://github.com/CQCL/tket/pull/743/commits/e06543a3343027c8ff962d77fda395d3b22d9f96)